src/System.Windows.Forms/src/System/Windows/Forms/HelpProvider.cs

```diff
@@ -80,9 +79,7 @@ public virtual string GetHelpKeyword(Control ctl)
         public virtual HelpNavigator GetHelpNavigator(Control ctl)
         {
             ArgumentNullException.ThrowIfNull(ctl);
-
-            object nav = _navigators[ctl];
-            return nav is null ? HelpNavigator.AssociateIndex : (HelpNavigator)nav;
+            return _navigators.TryGetValue(ctl, out HelpNavigator value) ? value : HelpNavigator.AssociateIndex;
```

Comment:
This needs to allow for a null `HelpNavigator`. Without null annotations to ensure that the `Dictionary` doesn't allow nulls, play safe.

```suggestion
            return _navigators.TryGetValue(ctl, out HelpNavigator value) && value is not null ? value : HelpNavigator.AssociateIndex;
```

Reply:
If I add that I get the following error: `Cannot convert null to 'HelpNavigator' because it is a non-nullable value type`.

Reply:
Sorry, thought it was a class, my mistake. :)